src/linear_algebra/bilinear_map.lean

```diff
@@ -31,34 +31,56 @@ namespace linear_map
 section semiring
 
 variables {R : Type*} [semiring R] {S : Type*} [semiring S]
+variables {R₂ : Type*} [semiring R₂] {S₂ : Type*} [semiring S₂]
 variables {M : Type*} {N : Type*} {P : Type*}
+variables {Nₗ : Type*} {Pₗ : Type*}
 variables {M' : Type*} {N' : Type*} {P' : Type*}
 
 variables [add_comm_monoid M] [add_comm_monoid N] [add_comm_monoid P]
+variables [add_comm_monoid Nₗ] [add_comm_monoid Pₗ]
 variables [add_comm_group M'] [add_comm_group N'] [add_comm_group P']
-variables [module R M] [module S N] [module R P] [module S P]
-variables [module R M'] [module S N'] [module R P'] [module S P']
-variables [smul_comm_class S R P] [smul_comm_class S R P']
-include R
+variables [module R M] [module S N] [module R₂ P] [module S₂ P]
+variables [module R Pₗ] [module S Pₗ]
+variables [module R M'] [module S N'] [module R₂ P'] [module S₂ P']
+variables [smul_comm_class S₂ R₂ P] [smul_comm_class S R Pₗ] [smul_comm_class S₂ R₂ P']
+variables {ρ₁₂ : R →+* R₂} {σ₁₂ : S →+* S₂}
+--include R
 
 variables (R S)
 /-- Create a bilinear map from a function that is linear in each component.
 See `mk₂` for the special case where both arguments come from modules over the same ring. -/
-def mk₂' (f : M → N → P)
+def mk₂' (f : M → N → Pₗ)
```

Comment:
Can you implement this in terms of `mk₂'ₛₗ` to make it clear that it's a special case?